scripts/production/homology_dump_cleanup.py

```diff
@@ -0,0 +1,265 @@
+                                                shutil.rmtree(dirs_to_remove)
+                                                dirs_removed = True
+                                            except Exception as err:
+                                                logging.error(f"Error removing directory: {dirs_to_remove}: {err}")
+                                                raise         
+                                        else:
+                                            logging.info(f"Dry run mode: Would have removed directory: {dirs_to_remove}")
+                            
+                            if not dirs_removed and not dry_run:
+                                logging.info("No directories found for removal based on the specified criteria.")
+
+          
+def main():
+    """
+    Entry point for the homology dumps cleanup script.
+
+    This function performs the following tasks:
+    1. Parses command-line arguments to get the homology dumps directory, master database URL, 
+       release cutoff, dry run option, and log file path.
+    2. Configures logging based on the provided log file path, if any.
+    3. Retrieves collections and division information from the master database using the provided URL.
+    4. Calls the cleanup_homology_dumps function to perform the cleanup based on the parsed arguments.
+
+    Command-line Arguments:
+        --homology_dumps_dir (str): The root directory of the homology dumps. Required.
+        --master_db_url (str): The URL of the master database. Optional.
+        --before_release (int): The Ensembl release cutoff for cleanup (non-inclusive). Required.
+        --dry_run: Perform a dry run without deleting files. Optional.
+        --log (str): Optional log file to record deleted files.
+
+    Returns:
+        None
+    """
+    # Create the parser
+    parser = argparse.ArgumentParser(description='Homology dumps cleanup script')
+
+    parser.add_argument('--homology_dumps_dir', type=str, required=True, help='Root directory of the homology dumps.')
+    parser.add_argument('--master_db_url', type=str, required=False, help='URL of the master database.')
+    parser.add_argument('--before_release', type=int, required=True, help='Ensembl release cutoff for cleanup (non-inclusive).')
+    parser.add_argument('--dry_run', action='store_true', help='Perform a dry run without deleting files.')
+    parser.add_argument('--log', type=str, help='Optional log file to record deleted files.')
+
+    args = parser.parse_args()
+    logging.basicConfig(filename=args.log, level=logging.INFO, format='%(asctime)s - %(message)s')
+
+    # Get collections and division info from master database
+    global collections, div_info
+
+    #Update this query to set a cutoff release before which collections should be ignored
+    #Update this query to have a placeholder for current_release where it is 113
+    collections_query = "SELECT DISTINCT ssh.name " \
```

Comment:
The queries might be better placed in the appropriate functions for clarity.

Reply:
Hi Botond, so both these queries use the same function for mysql connection and running the query, which is why I did not want to include the queries in the function. So if it is alright with you I would like to not include the queries in the function.